### PR TITLE
[9.x] SOLR-16943 Move Jetty HttpClient tracing into InstrumentedHttpListenerFactory

### PR DESCRIPTION
**Pull request by stillalex**
https://issues.apache.org/jira/browse/SOLR-16943

# Description

This is the 9.x equivalent of https://github.com/apache/solr/pull/1856

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by stillalex** [comment]
@dsmiley would appreciate your review on this backport to 9.x

File: solr/core/src/java/org/apache/solr/util/tracing/SolrRequestCarrier.java
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/util/tracing/SolrRequestCarrier.java:1]: This is now for a Jetty Request, not a SolrRequest; the class name and variable names and javadoc all need to be updated accordingly.

File: solr/core/src/java/org/apache/solr/util/tracing/SolrHttpRequestCarrier.java
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/util/tracing/SolrHttpRequestCarrier.java:1]: This is for an Apache HttpRequest.  Lets make that clear.

Comment (stillalex) [reply solr/core/src/java/org/apache/solr/util/tracing/SolrHttpRequestCarrier.java:1]: class renamed

File: solr/core/src/java/org/apache/solr/util/tracing/SolrRequestCarrier.java
Comment (stillalex) [reply solr/core/src/java/org/apache/solr/util/tracing/SolrRequestCarrier.java:1]: renamed

**Comment by stillalex** [comment]
did some manual testing and it looks ok

File: solr/core/src/java/org/apache/solr/util/tracing/SolrJettyRequestCarrier.java [hunk 1]
```diff
@@ -22,13 +22,13 @@
 import java.util.Map;
 import org.eclipse.jetty.client.api.Request;
 
-/** An OpenTracing Carrier for injecting Span context through SolrRequest */
-public class SolrRequestCarrier implements TextMap {
+/** An OpenTracing Carrier for injecting Span context through a Jetyy Request */
```
Comment (dsmiley) [inline solr/core/src/java/org/apache/solr/util/tracing/SolrJettyRequestCarrier.java:25]: typo

**Review by dsmiley — APPROVED** [review]

File: solr/core/src/java/org/apache/solr/util/tracing/SolrJettyRequestCarrier.java (hunk 1 above)
Comment (stillalex) [reply solr/core/src/java/org/apache/solr/util/tracing/SolrJettyRequestCarrier.java]: oups. fixed now. thanks for flagging

**Pull request merged**